Joss update (#46)
chrishalcrow/Skyrmions3D.jl: paper/paper.md

Reviewer: l38: > inefficies
inefficiencies?

The remarkable fact about the Skyrme model is that, despite its simplicity, the scattering and energy levels of skyrmions within the model qualitatively predict the observed properties of baryons when $F_\pi$, $e$ and $m_\pi$ are determined via experiment [@Adkins1983]. As a result, there is hope that studying the skyrme model (and simple modifications) can shed light into the theory of nuclei. For a comprehensive background on skyrmions and the surrounding literature see [@Manton2004], [@Manton2022]. 

> `Skyrmions3D.jl` implements a structure in Julia to describe numerically a skyrmion. It has

Reviewer 2: ```suggestion
`Skyrmions3D.jl` implements a structure in Julia to describe numerically a skyrmion. It contains
```

 
- the discrete grid $(x_i, y_j, z_k)$ of spatial values at which the Skyrme field will be given, 
- the pion field value at the points in the spatial grid, 
- the physical parameters $m_\pi$, $F_\pi$, $e$, and

> - boundary conditions determing how the Skyrme field is to be treated at the edge of the grid. 

Reviewer: > determing

determining?

Comprehensive documentation for `Skyrmions3D.jl` is provided via a webpage, including an API, examples of how to use key features, and guidance on requesting features or raising bug reports. The package also has a comprehensive unit-test suite.

# Software design

Julia was chosen for the implementation due to its balance between speed of execution, required especially for the computation of high-dimensional numerical integrals, and accessibility, both in terms of simplicity of coding and reproducibility of scientific results. As an example of the latter, there exist many helpful Julia packages such as `Pkg.jl` and `Documenter.jl` which ease the process of creating high-quality packages with simple reproducible installation instructions and clear documentation; both of these examples were used in the creation of `Skyrmions3D.jl`. That this decision has been successful is evidenced by the fact that papers implementing code within the `Skyrmions3D.jl` framework are already appearing [@Cork2025]. 

> By having the skyrmion structure implemented retain the information of the underlying spatial grid, pion field at all these points, as well as additional parameters, a single instance of a skyrmion often takes up a large amount of memory. This trade-off was accepted in order to allow the implementation of the saving and loading procedure of skymrions, a functionality which bolsters the reproducibilty of any scientific experiments performed with `Skyrmions3D.jl`. Additional steps were taken in the design of the package to boost its stability, such as the removal of interactive plotting functionality, which commonly interfered with other package depdencies.

Reviewer: I found the first sentence on this line difficult to parse, maybe rephrase to something like

> The implementation of the skyrmion structure in `Skyrmions3D.jl` retains the information of the underlying spatial grid, the pion field at these points, and several additional parameters; thus a single instance of a skyrmion often takes up a large amount of memory.

?

> The implementation of the skyrmion structure in `Skyrmions3D.jl` retains the information of the underlying spatial grid, the pion field at these points, and several additional parameters; thus a single instance of a skyrmion often takes up a large amount of memory. This trade-off was accepted in order to allow the implementation of the saving and loading procedure of skymrions, a functionality which bolsters the reproducibilty of any scientific experiments performed with `Skyrmions3D.jl`. Additional steps were taken in the design of the package to boost its stability, such as the removal of interactive plotting functionality, which commonly interfered with other package depdencies.

Reviewer 2: ```suggestion
The implementation of the skyrmion structure in `Skyrmions3D.jl` retains the information of the underlying spatial grid, the pion field at these points, and several additional parameters; thus a single instance of a skyrmion often takes up a large amount of memory. This trade-off was accepted in order to allow the implementation of the saving and loading procedure of skyrmions, a functionality which bolsters the reproducibility of any scientific experiments performed with `Skyrmions3D.jl`. Additional steps were taken in the design of the package to boost its stability, such as the removal of interactive plotting functionality, which commonly interfered with other package dependencies.
```

Also, I don't think this quite makes sense. What is the trade-off? Do you mean the skyrme field takes up a lot of space in memory (RAM)? Then the trade of is high memory use against fast computation. But this isn't to do with the saving/loading. The end of the paragraph is nice though!

Author: I had viewed the creation of a skyrmion structure as not being related to fast computation, as in principle one could implement functions to say compute the energy density which takes as an input the pion field, the grid, and the skyrme parameters. 

Instead I viewed this structure creation as a more fundamental principle about how to organise the information in a coherent way, making the project more scientific, and saw the save/load functionality as part of that. 

Have written some more words to try and expand upon this, is it clear, and would you agree?

Reviewer 2: Sounds good!

The underlying code was deliberately modularised to support future development, as it is known that modularity makes code easier to read, test, and refactor in later instances. 

# State of the field

> As previously written in the statement of need, there are no current alternatives to `Skyrmions3D.jl`, and this gives the software a unique relevance for researchers working on three dimensional skyrmions.

Reviewer 2: ```suggestion
There are no current alternatives to `Skyrmions3D.jl`, and this gives the software a unique relevance for researchers working on three dimensional skyrmions.
```

Don't think we need to refer to earlier bits of the paper.

> Moreover, while many of the underlying numerical methods in `Skyrmions3D.jl` are standard applications, there are notable exceptions which warrant special attention due to their wider significance for the topological solitions and differential geometry communities. The method used to approximate the holonomy of the instanton gauge field necessary for the Atiyah-Manton approximation is the only known implementation of [@Harland2023], developed specifically for the context of skyrmions, but which has wider applicability. Moreover, the ODE solved to flow an approximate skyrmion towards the (locally) minimal energy configuration is "arrested Newton flow", a modification of gradient flow common to the field of topological solitons but less widely known, see [@Battye2002], [@Gudnason2020]. The colouring used in plotting of skyrmion baryon density represents the dominant Skyrme field at that point in space, via the Runge colour sphere as introduced in [@Manton2012]. 

Reviewer: > solitions

solitons?

Future work on the package will focus on responding to the needs of the community. This may involve implementing new approximations for Skyrme fields, and by allowing for modifications of the standard Skyrme model. This further emphasizes the need to reproducible and refactorable code stated earlier. 

# Research Impact Statement

`Skyrmions3D.jl` is already having a tangible impact on the research of skyrmions, with a growing list of publications citing the package tracked in the software documentation. Moreover, the package has spurred future development with researchers independent of the core contributors creating forks of the code, and with new skyrmion software integrating `Skyrmions3D.jl` for its core functionality [@Cork2025]. 

# AI usage disclosure

> No generative AI tools were used in the development of this software, the writing of this manuscript, or the preparation of supporting materials. 

Reviewer 2: Woooo!
